From b22ab25991aba5ad71f26a68ea3b9366ac63ef4a Mon Sep 17 00:00:00 2001
From: Remi Cardona <remi@gentoo.org>
Date: Sat, 18 Oct 2008 01:01:23 +0200
Subject: [PATCH] force LANG=C when running awk due to unintended errors with some locales
MIME-Version: 1.0
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: 8bit

This bug was reported by a user trying to build the server with a
Turkish locale (tr_TR). The problem is that the Turkish alphabet is
latin-based, but not entirely similar. The bug comes from vesamodes
which has "Interlaced", which is then converted to lowercase by
modelines2c.awk. Execept that with a Turkish locale
tolower("Interlaced") is not "interlaced" but "Ä±nterlaced", which the
rest of the script fails to understand.

This patch forces LANG=C when running the awk script to always get the
intended latin en_US alphabet.
---
 hw/xfree86/common/Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/hw/xfree86/common/Makefile.am b/hw/xfree86/common/Makefile.am
index 0f44075..7f422c2 100644
--- a/hw/xfree86/common/Makefile.am
+++ b/hw/xfree86/common/Makefile.am
@@ -24,7 +24,7 @@ BUSSOURCES = xf86isaBus.c xf86pciBus.c xf86fbBus.c xf86noBus.c $(SBUS_SOURCES)
 MODEDEFSOURCES = $(srcdir)/vesamodes $(srcdir)/extramodes
 
 xf86DefModeSet.c: $(srcdir)/modeline2c.awk $(MODEDEFSOURCES)
-	cat $(MODEDEFSOURCES) | $(AWK) -f $(srcdir)/modeline2c.awk > $@
+	cat $(MODEDEFSOURCES) | LANG=C $(AWK) -f $(srcdir)/modeline2c.awk > $@
 
 BUILT_SOURCES = xf86DefModeSet.c
 
-- 
1.6.0.2

